parser_library/src/lsp/document_symbol_item.cpp

```diff
@@ -0,0 +1,44 @@
+
+
+#include "document_symbol_item.h"
+
+namespace hlasm_plugin::parser_library::lsp {
+
+document_symbol_item_s::document_symbol_item_s(hlasm_plugin::parser_library::context::id_index name,
+    document_symbol_kind kind,
+    range symbol_range,
+    range symbol_selection_range)
+    : name(name)
+    , kind(kind)
+    , symbol_range(symbol_range)
+    , symbol_selection_range(symbol_selection_range)
+{}
+document_symbol_item_s::document_symbol_item_s(
+    hlasm_plugin::parser_library::context::id_index name, document_symbol_kind kind, range symbol_range)
+    : name(name)
+    , kind(kind)
+    , symbol_range(symbol_range)
+    , symbol_selection_range(symbol_range)
+{}
+document_symbol_item_s::document_symbol_item_s(hlasm_plugin::parser_library::context::id_index name,
+    document_symbol_kind kind,
+    range symbol_range,
+    document_symbol_list_s children)
+    : name(name)
+    , kind(kind)
+    , symbol_range(symbol_range)
+    , symbol_selection_range(symbol_range)
+    , children(children)
+{}
+
+bool operator==(const document_symbol_item_s& lhs, const document_symbol_item_s& rhs)
+{
+    if (lhs.name == nullptr || rhs.name == nullptr)
+    {
+        return false;
+    }
+    return *(lhs.name) == *(rhs.name) && lhs.kind == rhs.kind && lhs.symbol_range == rhs.symbol_range
+        && lhs.symbol_selection_range == rhs.symbol_selection_range;
```

Comment:
It should be applied recursively on children. But by using the is_permutation_with_permutations function from the testing file. That is because for us the order of the document_symbol_items int he document_symbol_list does not matter.
In hindsight some unordered structure might have been more suitable than vector. Perhaps unordered set?

What is meant by scope?